android/app/src/main/java/org/fossasia/openevent/core/auth/profile/EditProfileActivity.java

```diff
@@ -82,7 +88,13 @@ protected void onCreate(Bundle savedInstanceState) {
             showUserData(user);
         });
 
-        avatar.setOnClickListener(v -> showFileChooser());
+        avatar.setOnClickListener(v -> {
+            if(!checkReadExternalPermission()) {
```

Comment:
instead of calling this function everytime keep a boolean to check if user has the permissions or not, update that boolean in the callback onRequestPermissionResult,
or lets wait for @iamareebjamal  for his views on this

Reply:
Okay

Reply:
Exactly, and take the reactive approach